app/admin/menu-items/edit/template.hbs

```diff
@@ -0,0 +1,8 @@
+{{#simple-form startingValues=model onsubmit=saveItem as |formValues|}}
```

Comment:
`saveItem` should be a string. Otherwise you are telling the simple form to run the action named whatever `saveItem` is set to as a variable. Since `saveItem` is undefined, it will not trigger anything.